    @Test
    public void removeAllFromRealm_invalidListShouldThrow() {
        RealmList<Dog> dogs = realm.where(Owner.class).findFirst().getDogs();
        assertEquals(TEST_SIZE, dogs.size());
        realm.close();
        realm = null;

        try {
            dogs.deleteAllFromRealm();
            fail("dogs is invalid and it should throw an exception");
        } catch (IllegalStateException e) {
            assertEquals("This Realm instance has already been closed, making it unusable.", e.getMessage());
        }
    }

